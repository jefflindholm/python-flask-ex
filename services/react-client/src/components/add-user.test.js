import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

import AddUser from './add-user';

test('AddUser renders properly', () => {
  const wrapper = shallow(<AddUser/>);
  const element = wrapper.find('form');
  expect(element.find('input').length).toBe(3);
  expect(element.find('input').get(0).props.name).toBe('username');
  expect(element.find('input').get(1).props.name).toBe('email');
  expect(element.find('input').get(2).props.type).toBe('submit');
});

test('AddUser renders snapshot properly', () => {
  const tree = renderer.create(<AddUser/>);
  expect(tree).toMatchSnapshot();
});
