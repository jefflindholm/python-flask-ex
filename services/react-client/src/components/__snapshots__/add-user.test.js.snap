// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddUser renders snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="username"
      onChange={undefined}
      placeholder="Enter username"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div
    className="field"
  >
    <input
      className="input is-large"
      name="email"
      onChange={undefined}
      placeholder="Enter email"
      required={true}
      type="text"
      value={undefined}
    />
  </div>
  <div>
    <input
      className="button is-primary is-large is-fullwidth"
      type="submit"
      value="Submit"
    />
  </div>
</form>
`;
